overlays/custom-packages/labwc/labwc-colored-borders.patch

```diff
@@ -0,0 +1,521 @@
++		//else  {
+ 		if (subtree == &ssd->border.active) {
+ 			color = theme->window_active_border_color;
++			wlr_scene_node_set_enabled(&parent->node, true);
+ 		} else {
+ 			color = theme->window_inactive_border_color;
+ 			wlr_scene_node_set_enabled(&parent->node, false);
+ 		}
++		//}
++
+ 		wl_list_init(&subtree->parts);
+ 		add_scene_rect(&subtree->parts, LAB_SSD_PART_LEFT, parent,
+ 			theme->border_width, height, 0, 0, color);
+diff --git a/src/ssd/ssd_part.c b/src/ssd/ssd_part.c
+index 4c72d29..9399435 100644
+--- a/src/ssd/ssd_part.c
++++ b/src/ssd/ssd_part.c
+@@ -80,12 +80,11 @@ add_scene_buffer(struct wl_list *list, enum ssd_part_type type,
+ 
+ struct ssd_part *
+ add_scene_button_corner(struct wl_list *part_list, enum ssd_part_type type,
+-		enum ssd_part_type corner_type, struct wlr_scene_tree *parent,
++		enum ssd_part_type corner_type, struct wlr_scene_tree *parent, float *bg_color,
+ 		struct wlr_buffer *corner_buffer, struct wlr_buffer *icon_buffer,
+ 		int x, struct view *view)
+ {
+ 	int offset_x;
+-	float invisible[4] = { 0, 0, 0, 0 };
+ 
+ 	if (corner_type == LAB_SSD_PART_CORNER_TOP_LEFT) {
+ 		offset_x = rc.theme->border_width;
+@@ -107,8 +106,8 @@ add_scene_button_corner(struct wl_list *part_list, enum ssd_part_type type,
+ 	add_scene_buffer(part_list, corner_type, parent, corner_buffer,
+ 		-offset_x, -rc.theme->border_width);
+ 
+-	/* Finally just put a usual theme button on top, using an invisible hitbox */
+-	add_scene_button(part_list, type, parent, invisible, icon_buffer, 0, view);
++	/* Finally just put a usual theme button on top, using an invisible/custom colored hitbox */
++	add_scene_button(part_list, type, parent, bg_color, icon_buffer, 0, view);
+ 	return button_root;
+ }
+ 
+diff --git a/src/ssd/ssd_titlebar.c b/src/ssd/ssd_titlebar.c
+index b0aaa2d..745e76b 100644
+--- a/src/ssd/ssd_titlebar.c
++++ b/src/ssd/ssd_titlebar.c
+@@ -24,6 +24,15 @@ ssd_titlebar_create(struct ssd *ssd)
+ {
+ 	struct view *view = ssd->view;
+ 	struct theme *theme = view->server->theme;
++
++	/* 1st approach: the whole theme changing */
++	struct theme custom_theme= { 0 };
++	float customColor[4];
++	if (window_rules_get_custom_border_color(view, customColor)) {
++		theme_customize_with_border_color(&custom_theme, customColor);
++		theme = &custom_theme;
++	}
++
+ 	int width = view->current.width;
+ 
+ 	float *color;
+@@ -43,6 +52,7 @@ ssd_titlebar_create(struct ssd *ssd)
+ 		subtree->tree = wlr_scene_tree_create(ssd->titlebar.tree);
+ 		parent = subtree->tree;
+ 		wlr_scene_node_set_position(&parent->node, 0, -theme->title_height);
++		
+ 		if (subtree == &ssd->titlebar.active) {
+ 			color = theme->window_active_title_bg_color;
+ 			corner_top_left = &theme->corner_top_left_active_normal->base;
+@@ -62,6 +72,18 @@ ssd_titlebar_create(struct ssd *ssd)
+ 			close_button_unpressed = &theme->button_close_inactive_unpressed->base;
+ 			wlr_scene_node_set_enabled(&parent->node, false);
+ 		}
++		
++		/* 2nd approach: colors changing
```

Comment:
See previous comment. If you want to leave some marks on the 2nd approach, please discuss in the markdown-document what was tried and why it did not work. Or why you ended up with the 1st approach: "the whole theme changing"